code/modules/jobs/jobs.dm

```diff
@@ -59,6 +59,21 @@ GLOBAL_LIST_INIT(nonhuman_positions, list(
 	"Cyborg",
 	"pAI"))
 
+GLOBAL_LIST_INIT(exp_jobsmap, list(

```

Comment:
if the job datums store their exp_type you should be able to autobuild this list from those definitions.